xmtp_mls/src/groups/subscriptions.rs

```diff
@@ -66,7 +66,8 @@ impl MlsGroup {
         );
 
         if let Some(GroupError::ReceiveError(_)) = process_result.as_ref().err() {
-            self.sync(&client).await?;
+            self.sync_with_conn(&client.mls_provider()?, &client)
```

Comment:
This change alone is going to resolve a lot of thrash and noise. Previously we were calling sync when we receive a commit from a stream. That would attempt to add missing members first.

Every member of the group who was online would try and do it at the same time, leading to a bunch of commits that would get thrown out.

We don't need to do this as part of stream processing.

Reply:
Nice catch. I'm thinking that another optimization would be to update `sync` so that if we sync during the `maybe_update_installations` process (`add_missing_installations` => `sync_until_intent_resolved`), we don't need to do a second `sync_with_conn` right afterwards.